features/support/env.rb

```diff
@@ -3,14 +3,28 @@
 require 'open4'
 require 'rspec'
 
+# rubocop:disable all
+
 SOURCE_DIRECTORY = "#{File.dirname(__FILE__)}/../../src"
 SHELL_OVERRIDE_DIRECTORY = "#{File.dirname(__FILE__)}/shell_overrides"
 
+MEMOIZED_REPOSITORY_BASE = Dir.mktmpdir 'memoized'
 REPOSITORY_BASE = Dir.mktmpdir
 TOOLS_INSTALLED_FILENAME = "#{REPOSITORY_BASE}/tools_installed.txt"
 
-Before do
+
+# copy entire contents of MEMOIZED_REPOSITORY_BASE to REPOSITORY_BASE
+def setup_environment
+  FileUtils.rm_rf Dir.glob("#{REPOSITORY_BASE}/*")
+  FileUtils.cp_r "#{MEMOIZED_REPOSITORY_BASE}/.", REPOSITORY_BASE
+
   Dir.chdir REPOSITORY_BASE
+  go_to_repository :developer
+end
+
+
+def memoize_environment
```

Comment:
I believe this should be called `initialize_environment`. The memoization is only a single line at the end of the function